src/09-tool-patterns/03-vanilla-tool-call.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Demonstrates that tools defined via Vercel AI SDK and
+ * can easily be called via the vendor's own API client and
+ * need not be locked into the Vercel abstractions.
+ *
+ * yarn tsx src/09-tool-patterns/03-vanilla-tool-call.ts
+ *
+ */
+import dotenv from "dotenv"
+import OpenAI from "openai"
+import dedent from "dedent"
+import { zodToJsonSchema } from "zod-to-json-schema"
+import { fetchShowsNearLocation } from "../lib/tools"
+
+dotenv.config()
+const openai = new OpenAI()
+
+async function main() {
+  // build initial message list
+
+  const messages: OpenAI.Chat.Completions.ChatCompletionMessageParam[] = [
+    {
+      role: "user",
+      content: dedent`
+        Give me 3 gallery shows happening in New York City"
+      `,
+    },
+  ]
+
+  console.log("Message:", messages)
+
+  // define tools, by accessing the AI SDK tool's attributes
+  // and shaping them for the vendor's API client
+
+  const tools: OpenAI.Chat.ChatCompletionTool[] = [
+    {
+      type: "function",
+      function: {
+        name: "fetchShowsNearLocation",
+        description: fetchShowsNearLocation.description,
+        parameters: zodToJsonSchema(fetchShowsNearLocation.parameters),
```

Comment:
The `tool` [object returned](https://github.com/artsy/quantum/pull/14/files#diff-3ac74bab8cef087a9f693269aee6a4ebdba8f544bed3911192e2bc011632109cR8) is easily adapted here to use with the OpenAI client directly…